File: ECommerceApi/ECommerceModels/ListRequest.cs
namespace ECommerceModels
{
    /// <summary>
    /// class for list request
    /// </summary>
    /// <typeparam name="T">the model which needs to be filtered</typeparam>
    public class ListRequest<T>
    {
        /// <summary>
        /// stores the page number
        /// </summary>
        public int PageNumber { get; set; }

        /// <summary>
        /// stores the page size
        /// </summary>
        public int PageSize { get; set; }

        /// <summary>
        /// stores the sort direction
        /// </summary>
        public string SortingDirection { get; set; }

        /// <summary>
        /// stores the sort field
        /// </summary>
        public string SortingField { get; set; }

        /// <summary>
        /// stores the filter model
        /// </summary>
        public T FilterModel { get; set; }
    }
}

File: LabSampleFrontEnd/src/models/list-request.ts
/**
 * class for list request
 */
export class ListRequest<T> {
    /**
     * page number
     */
    PageNumber: number;

    /**
     * page size
     */
    PageSize: number;

    /**
     * sorting direction
     */
    SortingDirection: string;

    /**
     * sorting field
     */
    SortingField: string;

    /**
     * defines the filter model
     */
    FilterModel: T;
}